scripts/compute_morphometrics_spine_generic.sh

```diff
@@ -0,0 +1,125 @@
+# 3. Runs a batch analysis (sct_run_batch) to compute the spinal cord cross-sectional area (CSA) on the test 
+#   subjects downloaded in step 2 using a given version of the contrast-agnostic model.
+# 4. Stores the output CSV files in the current directory, to be uploaded as part of the next release of the model
+# 
+# Usage:
+#   bash compute_morphometrics_spine_generic.sh
+
+
+# ==============================
+# DEFINE GLOBAL VARIABLES
+# ==============================
+# Path to the repository; all scripts will be relative to this path
+PATH_REPO="/home/GRAMES.POLYMTL.CA/${USER}/contrast-agnostic/contrast-agnostic-softseg-spinalcord"
+
+# Path to the output folder; the data, model, results, etc. will be stored in this folder
+PATH_OUTPUT="/home/GRAMES.POLYMTL.CA/${USER}/contrast-agnostic/test-post-training-script"
+
+# Path to the folder where the model exists, will be copied to the output folder PATH_OUTPUT
+# for testing purposes, replace the PATH_MODEL with the path to the model downloaded from the latest release
+PATH_MODEL="/home/GRAMES.POLYMTL.CA/${USER}/contrast-agnostic/sct_deployed_models/model_nnunet-AllRandInit3D"
+
+# NOTE: To be compatible previous releases of the models, and to be able to automatically generate the
+# morphometric plots after a new model is released, the model folder after copying will have the following
+# syntax: ${PATH_OUTPUT}/model_${VERSION_TO_BE_RELEASED}
+# Example: If the latest release in the contrast-agnostic repo points to the tag v3.1,
+# then the next version to be released is v3.2. 
+VERSION_TO_BE_RELEASED=3.2
+
+# Number of parallel processes to run (choose a smaller number as inference is run only on 1 gpu)
+NUM_WORKERS=4
+
+# ID of the GPU to run inference on {0,1,2,3}
+CUDA_DEVICE_ID=3
+
+# Create folders 
+mkdir -p "${PATH_OUTPUT}"
+mkdir -p "${PATH_OUTPUT}/model_v${VERSION_TO_BE_RELEASED}"
+
+# Copy the model to the output folder
+cp -r ${PATH_MODEL}/* ${PATH_OUTPUT}/model_v${VERSION_TO_BE_RELEASED}
+
+# Exit if user presses CTRL+C (Linux) or CMD+C (OSX)
+trap "echo Caught Keyboard Interrupt within script. Exiting now.; exit" INT
+
+# ==============================
+# DOWNLOAD DATA
+# ==============================
+
+echo "=============================="
+echo "Downloading test data ..."
+echo "=============================="
+
+# Clone the dataset and initialize a git annex repository
+git clone https://github.com/spine-generic/data-multi-subject ${PATH_OUTPUT}/data-multi-subject
+cd ${PATH_OUTPUT}/data-multi-subject
+
+# Initialize an empty git-annex repository 
+git annex init
+
+# Get the (frozen) test split of the spine-generic dataset to compute the morphometrics on
+# And download them using git-annex
+python3 -c 'import yaml, sys; 
+test_subjects = yaml.safe_load(open(sys.argv[1]))["test"]; 
+for subject in test_subjects: print(subject)' ${PATH_REPO}/scripts/spine_generic_test_split_for_csa_drift_monitoring.yaml | 
```

Comment:
I guess we could deduplicate this by using `TEST_SUBJECTS` as done before?

Reply:
great suggestion! fixed here https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/pull/139/commits/a2713324649ed997b6d060c86882fd541d4af7d1

Reply:
Would we be able to do the `readarray` command first, and then iterate over the array (rather than echoing)?

Reply:
ah I got your point now! fixed here https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/pull/139/commits/06160b8cf765ddb27287584f7841d671d93e75da

Reply:
Perfect! I think we can now remove the 2nd readarray command too?